### PR TITLE
Sample entropy

### PR DESCRIPTION
**Pull request by mynameisvinn**
wrt open request for sample entropy calculation (https://github.com/blue-yonder/tsfresh/issues/50), ive updated feature calculations and corresponding test.

### CODE REVIEW

**Comment by MaxBenChrist** [comment]
Hi @mynameisvinn, thank you for contributing again!

Unfortunately, some unit tests are failing :-(. Can you see the travis ouput? Did the tests pass locally on your machine?

**Comment by MaxBenChrist** [comment]
I think your merge failed? You still have `HEADS` and `+>>>>>>> upstream/master` lines on your code

File: tsfresh/feature_extraction/feature_calculators.py [hunk 1]
```diff
@@ -939,6 +939,61 @@ def binned_entropy(x, max_bins):
 
 @set_property("fctype", "aggregate_with_parameters")
 @not_apply_to_raw_numbers
+def sample_entropy(x, tolerance=None):
+    """
+    Calculate and return sample entropy of x.
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Would be nice to have a latex eformula of the feature here

File: tsfresh/feature_extraction/feature_calculators.py [hunk 2]
```diff
@@ -939,6 +939,61 @@ def binned_entropy(x, max_bins):
...
+    if tolerance is None:
+        tolerance = 0.1 * np.std(time_series)
+
+    n = len(time_series)
+    prev = np.zeros((1, n))
+    curr = np.zeros((1, n))
+    A = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length - 1]
+    B = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length]
+
+    for i in xrange(n - 1):
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Xrange is not supported in py3

File: tsfresh/feature_extraction/feature_calculators.py [hunk 3]
```diff
@@ -939,6 +939,61 @@ def binned_entropy(x, max_bins):
...
+        nj = n - i - 1
+        ts1 = time_series[i]
+        for jj in xrange(nj):
+            j = jj + i + 1
+            if abs(time_series[j] - ts1) < tolerance:  # distance between two vectors
+                curr[0, jj] = prev[0, jj] + 1
+                temp_ts_length = min(sample_length, curr[0, jj])
+                for m in xrange(int(temp_ts_length)):
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Did you try to vectorize parts or the whole algorithm?

File: tsfresh/feature_extraction/feature_calculators.py [hunk 4]
```diff
@@ -939,6 +939,61 @@ def binned_entropy(x, max_bins):
...
+    A = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length - 1]
+    B = np.zeros((sample_length, 1))  # number of matches for m = [1,...,template_length]
+
+    for i in xrange(n - 1):
+        nj = n - i - 1
+        ts1 = time_series[i]
+        for jj in xrange(nj):
+            j = jj + i + 1
+            if abs(time_series[j] - ts1) < tolerance:  # distance between two vectors
+                curr[0, jj] = prev[0, jj] + 1
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Curr and previous stay 1 dimensional the whole time, why make them two dimensional?

File: tsfresh/feature_extraction/feature_calculators.py (hunk 2 above)
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Just replace it by range

File: tsfresh/examples/har_dataset.py [hunk 5]
```diff
@@ -46,11 +62,24 @@ def download_har_dataset():
...
+<<<<<<< HEAD
+            zfile.extractall('data')
+=======
             zfile.extractall(path=data_file_name)
+>>>>>>> upstream/master
         zfile.close()
 
 
 def load_har_dataset():
+<<<<<<< HEAD
```
Comment (MaxBenChrist) [inline tsfresh/examples/har_dataset.py]: Merging fails here or?

**Comment by MaxBenChrist** [comment]
@mynameisvinn will you fix this in the near future? I WANT that feature + example 😀

**Comment by mynameisvinn** [comment]
@MaxBenChrist working on it!

**Pull request closed without merging**